src/arch/riscv/pagetable_walker.cc

```diff
@@ -670,10 +1071,29 @@ Walker::WalkerState::retry()
 }
 
 Fault
-Walker::WalkerState::pageFault(bool present)
+Walker::WalkerState::pageFault()
 {
-    DPRINTF(PageTableWalker, "Raising page fault.\n");
-    return walker->tlb->createPagefault(entry.vaddr, mode);
+    if (debug_condition()) {
+        DPRINTF(PageTableWalker,
+            "[%s] Raising page fault. vaddr: %#x gvaddr: %#x\n",
+            curstage == FIRST_STAGE ? "1stSTAGE" : "GSTAGE",
+            entry.vaddr, gresult.vaddr);
+    }
+    return walker->tlb->createPagefault(entry.vaddr, mode,
+                        gresult.vaddr, curstage == GSTAGE, memaccess.virt);
+}
+
+bool
+Walker::WalkerState::debug_condition() {
```

Comment:
This should be removed. If you want to print things in separately from the current debug flags, add a new one. You can also add compound debug flags. 

I can understand wanting to do this for a subset of addresses, but I don't think a one-off implementation in the ptw is the right way to do this.